core/services/AuthService.js

```diff
@@ -1,8 +1,26 @@
 const bcrypt = require("bcrypt");
 const config = require("config");
+const jwt = require("jsonwebtoken");
 
 module.exports = {
   encryptPassword: password => bcrypt.hashSync(password, config.get("bcryptSaltRounds")),
 
-  checkPassword: (password, passwordHash) => bcrypt.compareSync(password, passwordHash)
+  checkPassword: (password, passwordHash) => bcrypt.compareSync(password, passwordHash),
+
+  getOperatorByCredentials: async (email, password) => {
+    let operator = await yaxys.db.findOne ("operator", {email: email}, {});
+    if (operator && AuthService.checkPassword(password, operator.passwordHash)) {
+      return operator;
+    }
+    throw new Error("No such operator or wrong password provided");
+  },
+
+  generateToken: (operator) => jwt.sign(operator, config.get("jwt.secret"), {expiresIn: config.get("jwt.lifetime")}),
+
+  checkAndDecodeToken: (token) => {
+    let result = jwt.verify(token, config.get("jwt.secret"));
+    delete(result.iat);
+    delete(result.exp);
```

Comment:
it' better not to delete `exp` – it is required at client side for detecting whether the auth is expired. And, if you want to return object without some attributes, you can use lodash omit function: `_.omit(result, "iat", "some_other_unnecessary_field")`